torch/csrc/api/src/optim/adam.cpp

```diff
@@ -42,9 +42,8 @@ void AdamOptions::serialize(torch::serialize::InputArchive& archive) {
 bool operator==(const AdamParamState& lhs, const AdamParamState& rhs) {
   return (lhs.step() == rhs.step()) &&
           torch::equal(lhs.exp_avg(), rhs.exp_avg()) &&
-          torch::equal(lhs.exp_avg_sq(), rhs.exp_avg_sq()) &&
-          ((!lhs.max_exp_avg_sq().defined() && !rhs.max_exp_avg_sq().defined()) ||
-           (lhs.max_exp_avg_sq().defined() && rhs.max_exp_avg_sq().defined() && torch::equal(lhs.max_exp_avg_sq(), rhs.max_exp_avg_sq())));
+          torch::equal(lhs.exp_avg_sq(), rhs.exp_avg_sq());// &&
```

Comment:
hmm why do we have `// &&` at the end of this line?

Reply:
also I am not sure if there currently is `&&` operation between this line and the next line